services/provider/server/server.go

```diff
@@ -274,13 +274,15 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		return nil, fmt.Errorf("secret %s data fsid is empty", monSecret)
 	}
 
+	// TODO
+	// This is just a temporary fix to get the ceph client name. In the future, we'll change it and will not depend on string conditions.
 	extR = append(extR, &pb.ExternalResource{
 		Name: monSecret,
 		Kind: "Secret",
 		Data: mustMarshal(map[string]string{
-			"fsid":         fsid,
-			"mon-secret":   "mon-secret",
-			"admin-secret": "admin-secret",
+			"fsid":          fsid,
+			"mon-secret":    "mon-secret",
+			"ceph-username": fmt.Sprintf("%s-%s", "cephclient-health-checker", consumerResource.Name),
```

Comment:
You need to add `"ceph-secret"` as well which is the keyring of the `"ceph-username"`.